### PR TITLE
infoschema: rewrite SchemaSimpleTableInfos for infoschema v2 to avoid network

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Issue Number: ref #50959 

Problem Summary:

### What changed and how does it work?

HTTP API for schema table is a burden for infoschema v2 when there are a lot of tables.
In the past, we have change SchemaTable to SimpleSchemaTableInfos, i.e. from full `model.TableInfo` to `model.TableNameInfo` to reduce the cost.

SimpleSchemaTableInfos fetch the table name and id using meta API, fetch data from tikv.
As we store all the name->id mapping now, we can avoid the network API.

### Check List

Tests 

- [X] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/infoschema/infoschema_v2.go [hunk 1]
```diff
@@ -795,33 +795,34 @@ func (is *infoschemaV2) SchemaSimpleTableInfos(ctx context.Context, schema model
 		return nil, nil // something wrong?
 	}
 
-retry:
-	dbInfo, ok := is.SchemaByName(schema)
-	if !ok {
+	// Ascend is much more difficult than Descend.
```
Comment (lance6716) [inline pkg/infoschema/infoschema_v2.go:798]: what do you mean by `difficult`, difficult to implement?

**Review by lance6716 — APPROVED** [review]

File: pkg/infoschema/infoschema_v2.go (hunk 1 above)
Comment (tiancaiamao) [reply pkg/infoschema/infoschema_v2.go:798]: Handle it in Ascend order is more difficult than handle it in Descend order.

The data organize like this:

```
| t1                      | t2     | t3    |
| v5  v6  v7  v8 v10  v12 | v7 v13 | v2 v5 |
```

If we want infoschma snapshot at v10, the result should be `[t1.v10     t2.v7]`

In Descend order, the total test condition could be:

```
if is.infoSchema.schemaMetaVersion >= item.schemaVersion {
    if curr == nil || curr.tableName != tableItems[i].tableName {
          // add curr to the result
    }
}
```

But in Ascend order, you may have a try to see how much more difficult

 :-)

**Review by wjhuang2016 — APPROVED** [review]

**Pull request merged**